### PR TITLE
SDK-2380 Update SDKs and switch to OpenUPM for EDM4U dependency

### PR DESCRIPTION
**Pull request by echo-branch**
## Reference
SDK-2380 Update EDM4U dependency handling
SDK-2324 Expose DMA APIs

## Summary
Migrate from bundled External Dependency Manager for Unity, to the OpenUPM. 
Also bump version to 2.0.0 to indicate the change in integration method.
Expose DMA APIs

## Motivation
External Dependency Manager for Unity recommends no longer bundling and using the OpenUPM distribution.
Also updated to newer iOS and Android SDKs.

## Type Of Change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Testing Instructions
Branch Testbed now requires EDM4U be installed via OpenUPM or similar.
https://github.com/googlesamples/unity-jar-resolver

See notion page on testing with Branch Testbed.

cc @BranchMetrics/saas-sdk-devs for visibility.

### CODE REVIEW

**Comment by echo-branch** [comment]
Once I get the testbed in a better state I'll remove the WIP.

File: BranchUnityTestBed/Assets/Branch/Editor/BranchDependencies.xml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 <dependencies>
   <androidPackages>
-    <androidPackage spec="io.branch.sdk.android:library:5.8.0" />
+    <androidPackage spec="io.branch.sdk.android:library:5.11.0" />
```
Comment (gdeluna-branch) [inline BranchUnityTestBed/Assets/Branch/Editor/BranchDependencies.xml]: Can now be 5.12.0

File: BranchUnityTestBed/ProjectSettings/AndroidResolverDependencies.xml [hunk 2]
```diff
@@ -1,55 +1,16 @@
 <dependencies>
   <packages>
     <package>com.google.android.gms:play-services-ads-identifier:18+</package>
-    <package>io.branch.sdk.android:library:5.8.0</package>
+    <package>io.branch.sdk.android:library:5.11.0</package>
```
Comment (gdeluna-branch) [inline BranchUnityTestBed/ProjectSettings/AndroidResolverDependencies.xml]: 5.12.0

File: Changelog.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 Branch Unity SDK change log
 
+- 2.0.0
+ * External Dependency Manager for Unity (EDM4U) is no longer bundled with the Branch SDK. Clients will need to install it separately.
+ * Update iOS SDK to 3.4.2
+ * Update Android SDK to 5.11.0
```
Comment (gdeluna-branch) [inline Changelog.md]: 5.12.0

**Comment by gdeluna-branch** [comment]
Is there a ticket for the public documentation updates?

File: BranchUnityTestBed/Packages/manifest.json [hunk 4]
```diff
@@ -44,6 +44,16 @@
     "com.unity.modules.video": "1.0.0",
     "com.unity.modules.vr": "1.0.0",
     "com.unity.modules.wind": "1.0.0",
-    "com.unity.modules.xr": "1.0.0"
-  }
-}
+    "com.unity.modules.xr": "1.0.0",
+    "com.google.external-dependency-manager": "1.2.179"
```
Comment (gdeluna-branch) [inline BranchUnityTestBed/Packages/manifest.json:48]: So to confirm, the binaries were bundled in but now should be resolved by this package manager.

File: BranchUnityTestBed/ProjectSettings/AndroidResolverDependencies.xml [hunk 5]
```diff
@@ -1,55 +1,16 @@
...
+  <files />
   <settings>
     <setting name="androidAbis" value="armeabi-v7a" />
-    <setting name="bundleId" value="com.DefaultCompany.BranchUnityTestBed" />
+    <setting name="bundleId" value="io.branch.BranchUnityTestBed" />
     <setting name="explodeAars" value="True" />
     <setting name="gradleBuildEnabled" value="True" />
-    <setting name="gradlePropertiesTemplateEnabled" value="False" />
-    <setting name="gradleTemplateEnabled" value="False" />
+    <setting name="gradlePropertiesTemplateEnabled" value="True" />
```
Comment (gdeluna-branch) [inline BranchUnityTestBed/ProjectSettings/AndroidResolverDependencies.xml:12]: Does this change effectively mean to run gradle dependency management for Android? (Now that binaries are not included in the path by default)

File: BranchUnityTestBed/ProjectSettings/PackageManagerSettings.asset [hunk 6]
```diff
@@ -24,20 +25,20 @@ MonoBehaviour:
...
-      m_Capabilities: 0
     m_Modified: 0
-    m_Name: 
-    m_Url: 
-    m_Scopes:
-    - 
-    m_SelectedScopeIndex: 0
+    m_ErrorMessage: 
+    m_UserModificationsInstanceId: -828
+    m_OriginalInstanceId: -830
```
Comment (gdeluna-branch) [inline BranchUnityTestBed/ProjectSettings/PackageManagerSettings.asset:43]: Just curious where these numbers come from

File: BranchUnityTestBed/Packages/manifest.json (hunk 4 above)
Comment (gdeluna-branch) [reply BranchUnityTestBed/Packages/manifest.json:48]: In other words, this is what clients will now have to add in their integration

Comment (echo-branch) [reply BranchUnityTestBed/Packages/manifest.json:48]: Our SDK binaries were not bundled in. What was bundled in was the tool to fetch them, external dependency manager for unity. This is now what the client must install on their own.

File: BranchUnityTestBed/ProjectSettings/AndroidResolverDependencies.xml (hunk 5 above)
Comment (echo-branch) [reply BranchUnityTestBed/ProjectSettings/AndroidResolverDependencies.xml:12]: This isn't a hand written file. I believe it's generated by the external dependency manager for unity. The test app is checked in wholesale with all it's generated files.

File: BranchUnityTestBed/ProjectSettings/PackageManagerSettings.asset (hunk 6 above)
Comment (echo-branch) [reply BranchUnityTestBed/ProjectSettings/PackageManagerSettings.asset:43]: Same thing here. This is a generated file.

**Review by nsingh-branch — APPROVED** [review]

**Review by Nikki1980 — COMMENTED** [review]
@echo-branch

**Pull request merged**